cmd/kubelet/app/init_windows.go

```diff
@@ -34,26 +36,54 @@ const (
 // Ref: https://docs.microsoft.com/en-us/windows/win32/cimwin32prov/setpriority-method-in-class-win32-process
 func getPriorityValue(priorityClassName string) uint32 {
 	var priorityClassMap = map[string]uint32{
-		"IDLE_PRIORITY_CLASS":         uint32(64),
-		"BELOW_NORMAL_PRIORITY_CLASS": uint32(16384),
-		"NORMAL_PRIORITY_CLASS":       uint32(32),
-		"ABOVE_NORMAL_PRIORITY_CLASS": uint32(32768),
-		"HIGH_PRIORITY_CLASS":         uint32(128),
-		"REALTIME_PRIORITY_CLASS":     uint32(256),
+		"IDLE_PRIORITY_CLASS":         uint32(windows.IDLE_PRIORITY_CLASS),
+		"BELOW_NORMAL_PRIORITY_CLASS": uint32(windows.BELOW_NORMAL_PRIORITY_CLASS),
+		"NORMAL_PRIORITY_CLASS":       uint32(windows.NORMAL_PRIORITY_CLASS),
+		"ABOVE_NORMAL_PRIORITY_CLASS": uint32(windows.ABOVE_NORMAL_PRIORITY_CLASS),
+		"HIGH_PRIORITY_CLASS":         uint32(windows.HIGH_PRIORITY_CLASS),
+		"REALTIME_PRIORITY_CLASS":     uint32(windows.REALTIME_PRIORITY_CLASS),
 	}
 	return priorityClassMap[priorityClassName]
 }
 
+// addCurrentProcessToJobObjectAndSetPriorityClass creates a new Job Object
+// (https://docs.microsoft.com/en-us/windows/win32/procthread/job-objects),
+// adds the current process to the job object, and specifies the priority
+// class for the job object to the specified value.
+// A job object is used here so that any spawned processes such as powershell or
+// wmic are created at the specified thread priority class.
+// Running kubelet with above normal / high priority  can help improve
+// responsiveness on machines with high CPU utilization.
+func addCurrentProcessToJobObjectAndSetPriorityClass(pc uint32) error {
```

Comment:
This is a bit of a mouth full, but honestly I don't know what's idiomatic here or what's "too long". Maybe this could just set up the job (create + setinformation to the pc) and return the handle. Then you can just add the current process to it in initForOS right after.

Reply:
Updated, PTAL